cocoapods-plugin/lib/cocoapods-xcremotecache/command/hooks.rb

```diff
@@ -89,61 +89,67 @@ def self.generate_rcinfo()
-            config.build_settings['LD'] = ["#{xc_location}/xcld"]
-
-            config.build_settings['XCREMOTE_CACHE_FAKE_SRCROOT'] = FAKE_SRCROOT
-            add_cflags!(config.build_settings, '-fdebug-prefix-map', "#{user_proj_directory}=$(XCREMOTE_CACHE_FAKE_SRCROOT)")
-            add_swiftflags!(config.build_settings, '-debug-prefix-map', "#{user_proj_directory}=$(XCREMOTE_CACHE_FAKE_SRCROOT)")
+      def self.enable_xcremotecache(target, xc_location, xc_cc_path, mode, exclude_build_configurations, final_target)
+        target.build_configurations.each do |config|
+          # apply only for relevant Configurations
+          next if exclude_build_configurations.include?(config.name)
+          if mode == 'consumer'
+            config.build_settings['CC'] = ["$SRCROOT/#{xc_cc_path}"]
           end
-        
+          config.build_settings['SWIFT_EXEC'] = ["$SRCROOT/#{xc_location}/xcswiftc"]
+          config.build_settings['LIBTOOL'] = ["$SRCROOT/#{xc_location}/xclibtool"]
+          config.build_settings['LD'] = ["$SRCROOT/#{xc_location}/xcld"]
 
-        # User project is not generated from scratch (contrary to `Pods`), delete all previous XCRemoteCache phases
-        target.build_phases.delete_if {|phase| 
-          # Some phases (e.g. PBXSourcesBuildPhase) don't have strict name check respond_to?
-          if phase.respond_to?(:name) 
-              phase.name != nil && phase.name.start_with?("[XCRC]")
-          end
-        }
+          config.build_settings['XCREMOTE_CACHE_FAKE_SRCROOT'] = FAKE_SRCROOT
+          add_cflags!(config.build_settings, '-fdebug-prefix-map', "$SRCROOT=$(XCREMOTE_CACHE_FAKE_SRCROOT)")
+          add_swiftflags!(config.build_settings, '-debug-prefix-map', "$SRCROOT=$(XCREMOTE_CACHE_FAKE_SRCROOT)")
+        end
 
         # Prebuild
         if mode == 'consumer'
-          prebuild_script = target.new_shell_script_build_phase("[XCRC] Prebuild")
+          existing_prebuild_script = target.build_phases.detect do |phase|
+            if phase.respond_to?(:name)
+              phase.name != nil && phase.name.start_with?("[XCRC] Prebuild")
+            end
+          end
+          prebuild_script = existing_prebuild_script || target.new_shell_script_build_phase("[XCRC] Prebuild")
           prebuild_script.shell_script = "\"$SCRIPT_INPUT_FILE_0\""
-          prebuild_script.input_paths = ["#{xc_location}/xcprebuild"]
+          prebuild_script.input_paths = ["$SRCROOT/#{xc_location}/xcprebuild"]
           prebuild_script.output_paths = [
-            "$(TARGET_TEMP_DIR)/rc.enabled", 
+            "$(TARGET_TEMP_DIR)/rc.enabled",
             "$(DWARF_DSYM_FOLDER_PATH)/$(DWARF_DSYM_FILE_NAME)"
           ]
           prebuild_script.dependency_file = "$(TARGET_TEMP_DIR)/prebuild.d"
 
           # Move prebuild (last element) to the first position (to make it real 'prebuild')
-          target.build_phases.rotate!(-1)
+          target.build_phases.rotate!(-1) if existing_prebuild_script.nil?
```

Comment:
somewhere here we also need to delete `"[XCRC] Mark"` (if exists). We want to support a case when someone switches between producer -> consumer modes.